### PR TITLE
[v8] [browserstack-service] Fix E2Bug exception on linux

### PR DESCRIPTION
**Pull request by sriteja777**
## Proposed changes
Redact extensions from capabilities from setting into env, as this can be huge base64 string causing E2BIG exceptions on linux

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by sriteja777** [comment]
@christian-bromann Please review in your free time.

File: packages/wdio-browserstack-service/src/crash-reporter.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import type { BrowserstackConfig, CredentialsForCrashReportUpload, UserConfigfor
 import { DEFAULT_REQUEST_CONFIG, getObservabilityKey, getObservabilityUser } from './util.js'
 
 const log = logger('@wdio/browserstack-service')
+type Dict = {[key: string]: any};
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/crash-reporter.ts:10]:
```suggestion
type Dict = Record<string, any>
```

File: packages/wdio-browserstack-service/src/crash-reporter.ts [hunk 2]
```diff
@@ -83,13 +85,36 @@ export default class CrashReporter {
         })
     }
 
+    static recursivelyRedactKeysFromObject(obj: Dict|Array<Dict>, keys: string[]) {
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/crash-reporter.ts:88]:
```suggestion
    static recursivelyRedactKeysFromObject(obj?: Dict | Array<Dict>, keys: string[]) {
```

**Review by christian-bromann — APPROVED** [review]
👍 LGTM

**Comment by sriteja777** [comment]
@christian-bromann Have fixed it. Was happening due to [this](https://github.com/webdriverio/webdriverio/pull/11269/commits/7ca5cb7f2ea6874cc8aa45f2312ab07f778d58b9)

**Comment by grandEL-dsouza** [comment]
@christian-bromann 

I'm facing a similar issue. 

as a user please let me know what are the changes I should make to my repository post you deploy the fix. 

Thanks.

**Comment by Paren-theses** [comment]
Can confirm this fix is working for us using a linux box and an extension in our capabilities. TY!

**Comment by grandEL-dsouza** [comment]
@Paren-theses 

Thank you for your insight here.

**Pull request merged**